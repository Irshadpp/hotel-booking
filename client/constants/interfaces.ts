export interface User {
  id: string;
  email: string;
}

export interface Hotel {
  id: string;
  name: string;
  location: string;
  price: number;
}
