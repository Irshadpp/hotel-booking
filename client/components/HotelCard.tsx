import React, { useState } from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import { Hotel } from "@/constants/interfaces";

export default function HotelCard({
    hotel,
    handleBookNow,
  }: {
    hotel: Hotel;
    handleBookNow: (hotelId: string) => void;
  }) {
    return (
      <View key={hotel.id} style={styles.card}>
        <Text style={styles.name}>{hotel.name}</Text>
        <Text style={styles.location}>{hotel.location}</Text>
        <Text style={styles.price}>â‚¹{hotel.price}/day</Text>
        <TouchableOpacity
          style={styles.bookButton}
          onPress={() => handleBookNow(hotel.id)}
        >
          <Text style={styles.bookButtonText}>Book Now</Text>
        </TouchableOpacity>
      </View>
    );
  }

const styles = StyleSheet.create({
  card: {
    width: "100%",
    maxWidth: 350,
    backgroundColor: "#fff",
    padding: 20,
    marginVertical: 10,
    marginHorizontal: "auto",
    borderRadius: 15,
    shadowColor: "#000",
    shadowOpacity: 0.1,
    shadowOffset: { width: 0, height: 5 },
    shadowRadius: 10,
    elevation: 10, // Shadow effect for Android
  },
  name: {
    fontSize: 22,
    fontWeight: "bold",
    color: "#333",
    marginBottom: 5,
  },
  location: {
    fontSize: 16,
    color: "#555",
    marginBottom: 10,
  },
  price: {
    fontSize: 18,
    fontWeight: "bold",
    color: "#5fa35f",
    marginBottom: 20,
  },
  bookButton: {
    backgroundColor: "#5fa35f",
    paddingVertical: 10,
    borderRadius: 10,
    alignItems: "center",
  },
  bookButtonText: {
    fontSize: 16,
    color: "#fff",
    fontWeight: "bold",
  },
});
