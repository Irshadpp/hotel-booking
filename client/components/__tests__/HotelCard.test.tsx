import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { Hotel } from "@/constants/interfaces";
import HotelCard from "../HotelCard";

describe("HotelCard Component", () => {
  const mockHandleBookNow = jest.fn();
  const mockHotel: Hotel = {
    id: "1",
    name: "Grand Hotel",
    location: "New York, USA",
    price: 5000,
  };

  it("renders hotel details correctly", () => {
    const { getByText } = render(
      <HotelCard hotel={mockHotel} handleBookNow={mockHandleBookNow} />
    );

    expect(getByText("Grand Hotel")).toBeTruthy();
    expect(getByText("New York, USA")).toBeTruthy();
    expect(getByText("â‚¹5000/day")).toBeTruthy();
  });

  it('renders the "Book Now" button', () => {
    const { getByText } = render(
      <HotelCard hotel={mockHotel} handleBookNow={mockHandleBookNow} />
    );

    const button = getByText("Book Now");
    expect(button).toBeTruthy();
  });

  it("calls handleBookNow with the correct hotel ID when 'Book Now' is pressed", () => {
    const { getByText } = render(
      <HotelCard hotel={mockHotel} handleBookNow={mockHandleBookNow} />
    );

    const button = getByText("Book Now");
    fireEvent.press(button);

    expect(mockHandleBookNow).toHaveBeenCalledTimes(1);
    expect(mockHandleBookNow).toHaveBeenCalledWith("1");
  });

  it("matches the snapshot", () => {
    const tree = render(
      <HotelCard hotel={mockHotel} handleBookNow={mockHandleBookNow} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
