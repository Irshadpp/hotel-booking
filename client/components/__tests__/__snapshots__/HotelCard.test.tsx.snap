// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelCard Component matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 15,
      "elevation": 10,
      "marginHorizontal": "auto",
      "marginVertical": 10,
      "maxWidth": 350,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Grand Hotel
  </Text>
  <Text
    style={
      {
        "color": "#555",
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    New York, USA
  </Text>
  <Text
    style={
      {
        "color": "#5fa35f",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    â‚¹
    5000
    /day
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5fa35f",
        "borderRadius": 10,
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Book Now
    </Text>
  </View>
</View>
`;
